src/gc/gc.cpp

```diff
@@ -2963,8 +2966,12 @@ gc_heap::dt_low_ephemeral_space_p (gc_tuning_point tp)
             
             dprintf (GTC_LOG, ("h%d: plan eph size is %Id, new gen0 is %Id", 
                 heap_number, plan_ephemeral_size, new_gen0size));
-            ret = ((size_t)(heap_segment_reserved (ephemeral_heap_segment) - (heap_segment_mem (ephemeral_heap_segment))) <
-                    (plan_ephemeral_size + new_gen0size));
+
+            // While we were in the no gc region we could have allocated a larger than normal segment,
```

Comment:
We got here after we've *exited* from the no gc region mode, now we need to handle the seg we allocated "While we were in the no gc region".